internal/net/httpserver.go

```diff
@@ -127,16 +129,24 @@ func (h *HTTPServer) Stop() error {
 	return serveErr
 }
 
-func (h *HTTPServer) closeListener() (wasRunning bool, err error) {
+func (h *HTTPServer) shutdownServer() (wasRunning bool, _ error) {
 	h.lock.Lock()
 	defer h.lock.Unlock()
 
 	if h.listener == nil {
-		return
+		return false, nil
 	}
 
-	wasRunning = true
-	err = h.listener.Close()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
I'd really prefer to thread a configuration value through to this point, but that's more than I'd expect in this PR.

Fellow YARPCers, what do you think about landing this as-is and following up by threading a config value through? This is all internal API, so we can change the `NewHTTPServer` function to take a shutdown timeout, then make the corresponding changes in the HTTP config.